src/suricata.c

```diff
@@ -2778,6 +2779,16 @@ int main(int argc, char **argv)
 
     (void)SCSetThreadName("Suricata-Main");
 
+    /* Ignore SIGUSR2 as early as possble. We redeclare interest
+     * once we're done launching threads. The goal is to either die
+     * completely or handle any and all SIGUSR2s correctly.
+     */
+    UtilSignalHandlerSetup(SIGUSR2, SIG_IGN);
```

Comment:
this fails on mingw (windows / #ifdef OS_WIN32). It might be nicer to create a little wrapper 'UtilSignalSetupUSR2' or something, that handles the #ifdef'ing internally.

Reply:
btw it fails because SIGUSR2 is undefined

Reply:
Out of curiosity, is there any idea for performing rule reload under windows? I don't have a Win machine I can currently test on, but I think there is no rule reload there, am I correct?

Reply:
there are 2 levels of windows support currently. Cygwin has USR2 support and reloads too (iirc). The newer MinGW based code (OS_WIN32 define) does not have USR2 (and unix socket) and has the rule reload code ifdef'd out

Reply:
In the end, I'd prefer not to create a wrapper, we are already dealing with a wrapper here and it may be a little bit overwhelming. However, I correct the patch. Will create a PR shortly